htdocs/core/actions_massactions.inc.php

```diff
@@ -736,6 +736,20 @@
 	}
 }
 
+if ($massaction == 'cancelorders')
+{
+	$orders = GETPOST('toselect', 'array');
+	foreach ($orders as $id_order)
```

Comment:
All cancel must be into the same database transaction (db->begin db->commit)
Take example into massaction 'validate'